test/distributed/fsdp/test_fsdp_tp_integration.py

```diff
@@ -314,6 +351,78 @@ def test_fsdp_tp_integration(self, tensor_parallel_size, cpu_offload):
         tp_fsdp_out = tp_fsdp_model(inp)
         self.assertEqual(fsdp_out, tp_fsdp_out)
 
+    @skip_if_lt_x_gpu(4)
+    def test_fsdp_tp_gradient_layout(self):
```

Comment:
@awgu I added some tests on this change so this PR should be ready for review too :)

Reply:
Thanks! I will review soon.